lib/exceptions.rb

```diff
@@ -170,6 +170,7 @@ class UpgradeFailed < MachineryError; end
     class BinaryDiffError < MachineryError; end
 
     class SshConnectionFailed < MachineryError; end
+    class SshConnectionDisrupted < SystemExit; end
```

Comment:
Ah, I See. In this case I think we should rather handle `SshConnectionDisrupted` differently. Using the raw `SystemExit` class for that seems a bit crude to me.